lib/stringify.cpp

```diff
@@ -88,8 +88,11 @@ cstring vprintf_format(const char* fmt_str, va_list ap) {
     if (size < 0)
         throw std::runtime_error("Error in vsnprintf");
     if (static_cast<size_t>(size) >= sizeof(buf)) {
+        va_list ap_copy;
+        va_copy(ap_copy, ap);
```

Comment:
I think that you need to wrap the va_copy/va_end around the FIRST call to vsnprintf (line 87), rather than the second -- the first call leaves `ap` pointing after all the arguments so the second call gets garbage.  Copying after the first call still leaves it pointing after the arguments.